### DISCUSSION TITLE
Empty ebpf import path in generated .go

### DISCUSSION THREAD

**Discussion by chent1996** [cilium/ebpf#1298] [Q&A] [answered]
I tried to build ebpf with bpf2go on both the x86 and arm platforms but failed with arm because of the missing ebpf import path

```
// Code generated by bpf2go; DO NOT EDIT.

package bpf_core

import (
	"bytes"
	_ "embed"
	"fmt"
	"io"

	""
) 
```
with error:

```
2024/01/08 18:42:59 bazel-out/aarch64-fastbuild/bin/bpf_core/go_prog/from_lxc_v4nat_v6nat_bpf.go:6:2: could not import bytes (open : no such file or directory)
2024/01/08 18:42:59 bazel-out/aarch64-fastbuild/bin/bpf_core/go_prog/from_lxc_v4nat_v6nat_bpf.go:7:4: could not import embed (open : no such file or directory)
2024/01/08 18:42:59 bazel-out/aarch64-fastbuild/bin/bpf_core/go_prog/from_lxc_v4nat_v6nat_bpf.go:9:2: could not import io (open : no such file or directory)
2024/01/08 18:42:59 bazel-out/aarch64-fastbuild/bin/bpf_core/go_prog/from_lxc_v4nat_v6nat_bpf.go:11:2: invalid import path (empty string)
```
At first, I thought there was something to do with **currentModule()** in **output.go** but I think the module is not used to generate the ebpf path.

Since it works fine on x86, I'm not so sure if is there a problem with bpf2go or with my environment.

**Comment by lmb** [comment]
Do you have the most recent version? This might be another bazel bug.

https://github.com/cilium/ebpf/blob/394be60ec8d5eaa817fdf248c4a01e465aa7a082/cmd/bpf2go/tools.go#L88-L90

**Reply by chent1996** [reply]
The problem is not with Bazel, it is a change introduced by go 1.18
https://github.com/golang/go/issues/51831

If bpf2go is built with file arguments like `go build -o bpf2go_arm64_debug ./cmd/bpf2go/*.go`, then **debug.ReadBuildInfo()** will return ok but with empty buildInfo.Main

**Reply by florianl** [reply]
Just out of curiosity: Can you reproduce the issue with Go 1.20?
Asking, as the minimum supported Go version is Go 1.20 at the moment - see https://github.com/cilium/ebpf/blob/839118d43a4c6501f9d022a4b42b9633a778bf14/go.mod#L3

**Reply by chent1996** [reply]
I'm reproducing this with go 1.21.5

**Reply by lmb** [reply, marked as answer]
I'd argue that this is a non-standard way of building a package. Why not do `go build -o bpf2go_arm64_debug ./cmd/bpf2go`?
